packages/auto-label/src/helper.ts

```diff
@@ -278,32 +278,33 @@ export function autoDetectLabel(
 // A mapping of languages to their file extensions
 import defaultExtensions from './extensions.json';
 
-function getLabelFromPathConfig(
-  filename: string,
-  config: LanguageConfig
-): string {
+function getLabelFromPathConfig(filename: string, config: PathConfig): string {
   // If user specified labels for discrete paths
-  let label: string | PathConfig = '';
   const dirs = filename.split('/');
-  let paths = config.paths!;
-  // If user set default label for entire drive, use that label
-  if ('.' in paths) {
-    label = paths['.'];
-  }
+  let label = '';
+  let exactMatch = false;
+
+  // There are 3 possiblities for configuration file:
+  // 1. There is an exact match of filename and configuration path.
+  // In this case, set the label to that configuration. However, keep
+  // searching through the rest of the configuration file to find if
+  // there are any deeper matches.
+  // 2. There is a nested array or object that may contain a deeper match.
+  // If so, search recursively through that object.
+  // 3. If an exact deeper match was never found, set the label as whatever is
+  // listed for the whole directory, '.'
   for (const dir of dirs) {
```

Comment:
This is the for-loop that doesn't make sense to me.

Reply:
I see what you're saying. However, this is copied from the original function/author, so this preserves past functionality. Changing this might break expectations for others.